arangod/Agency/AgencyComm.cpp

```diff
@@ -692,19 +692,44 @@ void AgencyCommManager::addEndpoint(std::string const& endpoint) {
   if (iter == _endpoints.end()) {
     LOG_TOPIC(DEBUG, Logger::AGENCYCOMM) << "using agency endpoint '"
                                          << normalized << "'";
-
     _endpoints.emplace_back(normalized);
   }
 }
 
-void AgencyCommManager::removeEndpoint(std::string const& endpoint) {
+void AgencyCommManager::updateEndpoints(std::vector<std::string> const& newEndpoints) {
   MUTEX_LOCKER(locker, _lock);
+  
+  std::set<std::string> currentSet;
+  auto iter = _endpoints.begin();
+  for (; iter != _endpoints.end(); ++iter) {
+    currentSet.emplace(*iter);
+  }
 
-  std::string normalized = Endpoint::unifiedForm(endpoint);
-
-  _endpoints.erase(
-    std::remove(_endpoints.begin(), _endpoints.end(), normalized),
-    _endpoints.end());
+  std::set<std::string> updatedSet;
```

Comment:
not really critical, but the `updatedSet` can computed before the mutex is acquired, thus reducing the time required for holding the mutex.